packages/models/src/token-generation-readmodel/token-generation-states-entry.ts

```diff
@@ -70,3 +70,127 @@ export const TokenGenerationStatesGenericClient =
 export type TokenGenerationStatesGenericClient = z.infer<
   typeof TokenGenerationStatesGenericClient
 >;
+
+// GSI projection types
+// Agreement
+export const TokenGenStatesConsumerClientWithGSIPKConsumerIdEServiceIdProjection =
+  TokenGenerationStatesConsumerClient.pick({
+    PK: true,
+    GSIPK_consumerId_eserviceId: true,
+    agreementState: true,
+    descriptorState: true,
+    descriptorAudience: true,
+    descriptorVoucherLifespan: true,
+  });
+export type TokenGenStatesConsumerClientWithGSIPKConsumerIdEServiceIdProjection =
+  z.infer<
+    typeof TokenGenStatesConsumerClientWithGSIPKConsumerIdEServiceIdProjection
+  >;
```

Comment:
Open to naming suggestions. This is clear, but way too long.
Same for other new types.